src/meshapi/tests/test_replay_join_records.py

```diff
@@ -0,0 +1,145 @@
+from unittest.mock import patch
+
+from django.core import management
+from django.test import TestCase
+
+from meshapi.models.install import Install
+from meshapi.util.join_records import JoinRecord, JoinRecordProcessor, s3_content_to_join_record
+
+MOCK_JOIN_RECORD_PREFIX = "join-record-test"
+
+
+# Integration test to ensure that we can fetch JoinRecords from an S3 bucket,
+# replay them into the join form, and insert the objects we expect.
+# We should test a happy case, a case when the JoinRecord is bad (like a JoinRecord
+# somehow from Russia), and maybe mock a MeshDB 500 that just ends in us putting
+# the data back.
+class TestReplayJoinRecords(TestCase):
+    def setUp(self) -> None:
+        p = JoinRecordProcessor()
+        p.flush_test_data()
+
+    def tearDown(self) -> None:
+        p = JoinRecordProcessor()
+        p.flush_test_data()
+
+    # This is just to make codecov happy
+    def test_s3_content_to_join_record(self):
+        sample_key = "dev-join-form-submissions/2024/11/01/11/33/49.json"
+        sample_s3_content = """{"first_name":"Jon","last_name":"Smith","email_address":"js@gmail.com","phone_number":"+1 585-475-2411","street_address":"197 Prospect Place","apartment":"1","city":"Brooklyn","state":"NY","zip_code":"11238","roof_access":true,"referral":"I googled it.","ncl":true,"trust_me_bro":false,"code":"201","replayed":0,"install_number":1002}"""
```

Comment:
nit: Might be nice to let the auto-formatter have a crack at this JSON blob by making it a python dict wrapped by `json.dumps()`